File: fm/apps/event/js/plugins/Traceback.js
//---------------------------------------------------------------------
// Traceback plugin
//---------------------------------------------------------------------
// Copyright (C) 2007-2013 The NOC Project
// See LICENSE for details
//---------------------------------------------------------------------
console.debug("Defining NOC.fm.event.plugins.Traceback");

Ext.define("NOC.fm.event.plugins.Traceback", {
    extend: "Ext.panel.Panel",
    title: "Traceback",
    app: null,
    autoScroll: true,
    bodyPadding: 4,

    updateData: function(data) {
        var me = this;
        me.update("<PRE>" + data.traceback + "</PRE>");
    }
});

File: fm/apps/event/plugins/traceback.py
# -*- coding: utf-8 -*-
##----------------------------------------------------------------------
## TracebackPlugin
##----------------------------------------------------------------------
## Copyright (C) 2007-2013 The NOC Project
## See LICENSE for details
##----------------------------------------------------------------------

## NOC modules
from base import EventPlugin


class TracebackPlugin(EventPlugin):
    name = "traceback"

    def get_data(self, event, config):
        r = {}
        if "traceback" in event.raw_vars:
            r["traceback"] = event.raw_vars["traceback"]
            r["plugins"] = [
                ("NOC.fm.event.plugins.Traceback", {})
            ]
        return r
